core/trino-main/src/main/java/io/trino/metadata/ResolvedFunction.java

```diff
@@ -27,122 +25,36 @@
-            @JsonProperty("id") FunctionId functionId,
-            @JsonProperty("functionKind") FunctionKind functionKind,
-            @JsonProperty("deterministic") boolean deterministic,
-            @JsonProperty("functionNullability") FunctionNullability functionNullability,
-            @JsonProperty("typeDependencies") Map<TypeSignature, Type> typeDependencies,
-            @JsonProperty("functionDependencies") Set<ResolvedFunction> functionDependencies)
+    public ResolvedFunction
     {
-        this.signature = requireNonNull(signature, "signature is null");
-        this.catalogHandle = requireNonNull(catalogHandle, "catalogHandle is null");
-        this.functionId = requireNonNull(functionId, "functionId is null");
-        this.functionKind = requireNonNull(functionKind, "functionKind is null");
-        this.deterministic = deterministic;
-        this.functionNullability = requireNonNull(functionNullability, "functionNullability is null");
-        this.typeDependencies = ImmutableMap.copyOf(requireNonNull(typeDependencies, "typeDependencies is null"));
-        this.functionDependencies = ImmutableSet.copyOf(requireNonNull(functionDependencies, "functionDependencies is null"));
+        requireNonNull(signature, "signature is null");
+        requireNonNull(catalogHandle, "catalogHandle is null");
+        requireNonNull(functionId, "functionId is null");
+        requireNonNull(functionKind, "functionKind is null");
+        requireNonNull(functionNullability, "functionNullability is null");
+        typeDependencies = ImmutableMap.copyOf(requireNonNull(typeDependencies, "typeDependencies is null"));
+        functionDependencies = ImmutableSet.copyOf(requireNonNull(functionDependencies, "functionDependencies is null"));
         checkArgument(functionNullability.getArgumentNullable().size() == signature.getArgumentTypes().size(), "signature and functionNullability must have same argument count");
     }
 
-    @JsonProperty
-    public BoundSignature getSignature()
-    {
-        return signature;
-    }
-
-    @JsonProperty
-    public CatalogHandle getCatalogHandle()
-    {
-        return catalogHandle;
-    }
-
-    @JsonProperty("id")
-    public FunctionId getFunctionId()
-    {
-        return functionId;
-    }
-
-    @JsonProperty("functionKind")
-    public FunctionKind getFunctionKind()
-    {
-        return functionKind;
-    }
-
-    @JsonProperty
-    public boolean isDeterministic()
-    {
-        return deterministic;
-    }
-
-    @JsonProperty
-    public FunctionNullability getFunctionNullability()
-    {
-        return functionNullability;
-    }
-
-    @JsonProperty
-    public Map<TypeSignature, Type> getTypeDependencies()
-    {
-        return typeDependencies;
-    }
-
-    @JsonProperty
-    public Set<ResolvedFunction> getFunctionDependencies()
-    {
-        return functionDependencies;
-    }
-
     public CatalogSchemaFunctionName getName()
```

Comment:
ResolvedFunction conversion to record adds "name" into JSON form.
You need explicit `@JsonIgnore` on this. You can link to https://github.com/airlift/airlift/pull/1141 in a TODO comment.